src/createServer.js

```diff
@@ -1,13 +1,208 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const expenses = [];
+  const users = [];
+  let expenseIdCounter = 1;
+  let userIdCounter = 1;
+  const app = express();
+
+  app.use(express.json());
+
+  app.get('/expenses', (req, res) => {
+    const { userId, category, from, to, categories } = req.query;
+    let filteredExpenses = expenses;
+
+    if (userId) {
+      filteredExpenses = filteredExpenses.filter(
+        (exp) => exp.userId === parseInt(userId)
+      );
+    }
+
+    if (category) {
+      filteredExpenses = filteredExpenses.filter(
+        (exp) => exp.category === category
+      );
+    }
+
+    if (categories) {
+      const categoryArray = Array.isArray(categories)
+        ? categories
+        : categories.split(',').map((cat) => cat.trim());
+
+      filteredExpenses = filteredExpenses.filter((exp) =>
+        categoryArray.includes(exp.category)
+      );
+    }
+
+    if (from && to) {
+      const start = new Date(from);
+      const end = new Date(to);
+
+      if (isNaN(start.getTime()) || isNaN(end.getTime())) {
+        return res.status(400).json({ message: 'Invalid date format' });
+      }
+
+      filteredExpenses = filteredExpenses.filter((exp) => {
+        const expData = new Date(exp.spentAt);
+
+        return expData >= start && expData <= end;
+      });
+    }
+    res.json(filteredExpenses);
+  });
+
+  app.get('/expenses/:id', (req, res) => {
+    const id = parseInt(req.params.id);
+    const expense = expenses.find((exp) => exp.id === id);
+
+    if (!expense) {
+      return res.status(404).json({ message: 'Expense not found' });
+    }
+    res.json(expense);
+  });
+
+  app.post('/expenses', (req, res) => {
+    const { title, amount, userId, category, note, spentAt } = req.body;
```

Comment:
Consider adding validation for `amount` to ensure it's a valid number and greater than zero.